src/drt/src/ta/FlexTA_init.cpp

```diff
@@ -242,70 +245,59 @@ void FlexTAWorker::initIroute_helper_generic_helper(frGuide* guide,
         if (iterm->getNet() != net) {
           continue;
         }
-        inst = iterm->getInst();
-        shiftXform = inst->getTransform();
+        frInst* inst = iterm->getInst();
+        dbTransform shiftXform = inst->getTransform();
         shiftXform.setOrient(dbOrientType(dbOrientType::R0));
-        instXform = inst->getUpdatedXform();
-        trueTerm = iterm->getTerm();
+        frMTerm* mterm = iterm->getTerm();
+        int pinIdx = 0;
+        for (auto& pin : mterm->getPins()) {
+          if (!pin->hasPinAccess()) {
+            continue;
+          }
+          frAccessPoint* ap
+              = (static_cast<frInstTerm*>(term)->getAccessPoints())[pinIdx];
+          if (ap == nullptr) {
+            continue;
+          }
+          Point bp = ap->getPoint();
+          shiftXform.apply(bp);
+          if (getRouteBox().intersects(bp)) {
+            wlen2 = isH ? bp.y() : bp.x();
+            return;
+          }
+          pinIdx++;
+        }
         break;
       }
-      // TODO FIXME
-      // BTerms don't have an inst*, so the if (trueTerm)... code below doesn't
-      // actually act on BTerms. This code can be removed without changing
-      // functionality, but it is being left commented because it appears to be
-      // a bug with the way BTerms are handled and it may give a clue to what
-      // the anticipated behavior should be. Note that the bTerm can still
-      // affect whether wlen2 is set to 0 or not after if (trueTerm)...
       case frcBTerm: {
-        auto bTerm = static_cast<frBTerm*>(term);
-        if (bTerm->getNet() != net) {
+        auto bterm = static_cast<frBTerm*>(term);
+        if (bterm->getNet() != net) {
           continue;
         }
-        // trueTerm = bterm;
+        for (auto& pin : bterm->getPins()) {
+          if (!pin->hasPinAccess()) {
+            continue;
+          }
+          for (auto& ap : pin->getPinAccess(0)->getAccessPoints()) {
+            Point bp = ap->getPoint();
+            if (getRouteBox().intersects(bp)) {
+              wlen2 = isH ? bp.y() : bp.x();
+              return;
+            }
+          }
+        }
         break;
       }
       default:
         break;
     }
-
-    if (trueTerm) {
-      int pinIdx = 0;
-      int pinAccessIdx = (inst) ? inst->getPinAccessIdx() : -1;
-      for (auto& pin : trueTerm->getPins()) {
-        frAccessPoint* ap = nullptr;
-        if (inst) {
-          ap = (static_cast<frInstTerm*>(term)->getAccessPoints())[pinIdx];
-        }
-        if (!pin->hasPinAccess()) {
-          continue;
-        }
-        if (pinAccessIdx == -1) {
-          continue;
-        }
-        if (ap == nullptr) {
-          continue;
-        }
-        Point apBp = ap->getPoint();
-        shiftXform.apply(apBp);
-        if (getRouteBox().intersects(apBp)) {
-          wlen2 = isH ? apBp.y() : apBp.x();
-          return;
-        }
-        pinIdx++;
-      }
-    };  // to do @@@@@
-    wlen2 = 0;
```

Comment:
I remember when I was hacking at this code, removing this `wlen2 = 0` changed the results. Are you sure you want to remove it?

Reply:
Good catch, let me restore that for now. I struggled to understand the wlen vs wlen2 code but we can clarify and simplify it in subsequent patches.

Reply:
I took another look at this, and I'm not sure how setting wlen2 to zero is ever correct. wlen2 is the x or y coordinate of the pin or access point. If we set it to zero then `assignIroute_getPinCost` will be huge:

```
frUInt4 FlexTAWorker::assignIroute_getPinCost(taPin* iroute, frCoord trackLoc)
{ 
  frUInt4 sol = 0;
  if (iroute->hasWlenHelper2()) {
    sol = abs(trackLoc - iroute->getWlenHelper2());
```

I do plan to clean this up (and create better names, wlen, wlen2 are just confusing).

Reply:
Given this change this will need to go through a full secure CI testing just to be safe.  Is this otherwise ready to go?

Reply:
The PR is ready. I've tried to decipher the logic here and I'm pretty sure it's a mistake to set wlen2 to 0.

Reply:
I've started the run.

Reply:
The run outcome is good